File: Leetcode/longest_increasing_subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        
        
        
         int []dp = new int[nums.length];
        
        
        int overAllMax = 0;
        
        for(int i=0; i<dp.length; i++){
            
            int max = 0;
            
            for(int j=0; j<i; j++){
                
                if(nums[j]<nums[i]){
                    
                        if(dp[j]>max){
                            max= dp[j];
                        }
                    
                }
            }
            
            dp[i] = max + 1;
            
            if(dp[i]>overAllMax){
                
                overAllMax=dp[i];
                
            }
            
        }
        
       return overAllMax;
        
    
        
        
    
    }
}

File: self/LongestIncreasingSubsequence.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int lengthOfLIS(vector<int> &nums)
    {
        int n = nums.size();
        vector<int> dp(n, 1);

        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < i; j++)
            {
                if (nums[j] < nums[i])
                {
                    dp[i] = max(dp[i], dp[j] + 1);
                }
            }
        }

        int ans = dp[0];
        for (int i = 1; i < n; i++)
        {
            ans = max(ans, dp[i]);
        }

        return ans;
    }
};